### PR TITLE
COMMERCE-10142 - SE -  We only need to check whether accountId is 0

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-10142

@andrea-ale-sbarra , assigned to you since Luca is on medical leave.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java [hunk 1]
```diff
@@ -88,11 +88,7 @@ public int doStartTag() throws JspException {
 
 			if ((cpSku != null) && !hasChildCPDefinitions) {
 				_cpInstanceId = cpSku.getCPInstanceId();
-				_disabled =
-					!cpSku.isPurchasable() ||
-					((_commerceAccountId <= 0) &&
-					 !_commerceOrderHttpHelper.isGuestCheckoutEnabled(
-						 httpServletRequest));
+				_disabled = !cpSku.isPurchasable() || (_commerceAccountId == 0);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java:91]: I think even in this case the button will be disabled for guest user. At this point I think the code should be like this:
````suggestion
_disabled = !cpSku.isPurchasable()
````

**Review by andrea-ale-sbarra — COMMENTED** [review]
@alessiorendina can you have a look at this too?

File: modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java (hunk 1 above)
Comment (alessiorendina) [reply modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java:91]: accountId 0 means that the user is signed in but it didn't select an account. So, I think it is correct to disable it in this scenario.

Comment (brianikim) [reply modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java:91]: @andrea-ale-sbarra , just to give context in B2C, a personal account is created as soon as user logs in. accountId will never be 0 in this case. In B2B, only time accountId 0 is user is signed in, but it did not select an account. acountId will always be -1 if Guest user: https://github.com/liferay-commerce/liferay-portal/blob/f81bedd689912175457afaef6b78108a50309007/modules/apps/account/account-api/src/main/java/com/liferay/account/constants/AccountConstants.java#L26

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java:91]: Oh yeah you're right IDK why I was convinced that GUEST was 0 and not -1

Thank you for the explanations!

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:commerce

**Comment by victoroscuro** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
forwarded here https://github.com/liferay-commerce/liferay-portal/pull/3230

**Pull request closed without merging**